x/staking/keeper/delegation.go

```diff
@@ -73,11 +78,16 @@ func (k Keeper) Withdraw(ctx sdk.Context, delAddr sdk.AccAddress, token sdk.DecC
 		return time.Time{}, types.ErrInsufficientDelegation(types.DefaultCodespace, quantity.String(), delegator.Tokens.String())
 	}
 
+	// proxy have to unreg before withdrawing total tokens
+	leftTokens := delegator.Tokens.Sub(quantity)
+	if delegator.IsProxy && leftTokens.IsZero() {
+		return time.Time{}, types.ErrInvalidProxyWithdrawTotal(types.DefaultCodespace, delAddr.String())
```

Comment:
move this code to the section of `if leftTokens.IsZero() {`, it make lessens code